subsys/net/lib/nrf_cloud/src/nrf_cloud_transport.c

```diff
@@ -15,7 +15,7 @@
 #include <misc/util.h>
 
 #if defined(CONFIG_BSD_LIBRARY)
-#include "nrf_inbuilt_key.h"
+#include <nrf_socket.h>
```

Comment:
Why do we need to include `nrf_socket.h` all of sudden? We have offloaded `net/socket.h` for socket functions.

Reply:
In case of `CONFIG_NRF_CLOUD_PROVISION_CERTIFICATES` is not set, functions like `nct_client_id_get` needs `nrf_socket.h`. I added it to be included if `CONFIG_BSD_LIBRARY` is set.

Reply:
Earlier we blindly included `nrf_inbuilt_key.h` which included `nrf_socket.h`. This is not the case anymore if you opt out certificate provisioning.

Reply:
I see, I wonder why is it still using AT socket directly instead of using at_cmd driver. But it's not an issue to solve in this PR.